### PR TITLE
feat: added possibilities to change theme

### PR DESCRIPTION
**Pull request by mukiienko**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. Use imperative, present tense in your commit description ("change", not "changed" or "changes") without uppercases or period (.) at the end.

# Checklist
- [x] The implementation has been manually tested and complies with Textkernel [browser support guidelines](https://textkernel.com/browser-support/)
- [x] The implementation complies with [accessibility](CONTIRBUTING.md#accessibility) standards.
- [x] The component has a [displayName](CONTRIBUTING.md#display-names) defined.
- [x] The component comes with a detailed [PropTypes](CONTRIBUTING.md#component-props) (and defaultProps) definition.
- [x] Component PropTypes are sufficiently described / documented.
- [x] There is [a story](CONTRIBUTING.md#component-showcases) in Storybook.

### CODE REVIEW

File: src/components/Themeroller/Themeroller.tsx [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
-import { ItemValue } from './ItemValue';
-import { useThemeConfig } from './useThemeConfig';
+import { Field } from '../Field';
+import { Input } from '../Input';
+import { Callout } from '../Callout';
+import { ThemeTuner } from './ThemeTuner';
+import { ThemeActions } from './ThemeActions';
 import styles from './Themeroller.scss';
 
 type CssVars = {
```
Comment (ahalyha) [inline src/components/Themeroller/Themeroller.tsx:16]: Maybe lets move it to constants? or remove

File: src/components/Themeroller/Themeroller.tsx [hunk 2]
```diff
@@ -18,72 +23,147 @@ export type ThemerollerChildrenProps = {
...
-const { elem } = bem('Themeroller', styles);
+const { block, elem } = bem('Themeroller', styles);
 
 export const Themeroller: React.FC<Props> = ({
-    themeConfig: initialThemeConfig,
-    children,
-    onGenerate,
+    config,
+    themeResultData,
+    inputLabel = '',
```
Comment (ahalyha) [inline src/components/Themeroller/Themeroller.tsx:50]: maybe lets move it to default props?

File: src/components/Themeroller/ThemeActions/ThemeActions.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import { FaDownload, FaUpload, FaStop } from 'react-icons/fa';
+import { bem } from '../../../utils';
+import { Button, FileButton } from '../../Buttons';
+import { Tooltip } from '../../Tooltip';
+import styles from './ThemeActions.scss';
+
+type Props = {
```
[the remark this answers is not in the document]
Comment (mukiienko) [reply src/components/Themeroller/ThemeActions/ThemeActions.tsx:8]: yep, we always use `interface` for `Props` in OneUI. I'll change.

File: src/components/Themeroller/ThemeActions/ThemeActions.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    downloadLabel,
+    fileLabel,
+    downloadTooltipLabel,
+    downloadDisabled,
+    onReset,
+    onDownload,
+    onFileChange,
+}) => {
+    const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        if (onFileChange && e.target && e.target.files && e.target.files[0]) {
```
[the remark this answers is not in the document]
Comment (mukiienko) [reply src/components/Themeroller/ThemeActions/ThemeActions.tsx:40]: Good point

File: stories/Themeroller.tsx [hunk 5]
```diff
@@ -1,86 +1,30 @@
...
-            OneUI.applyThemeStyle(css);
+        const handleChange = (themeResult) => {
+            sessionStorage.setItem(OneUI.THEME_ELEMENT_ID, JSON.stringify(themeResult));
         };
-
+        let themeResultData;
+        try {
+            themeResultData = JSON.parse(sessionStorage.getItem(OneUI.THEME_ELEMENT_ID) || '');
+        } catch (err) {
+            console.log(err);
```
[the remark this answers is not in the document]
Comment (mukiienko) [reply stories/Themeroller.tsx:16]: Replaced to error

File: src/components/Themeroller/Themeroller.tsx [hunk 6]
```diff
@@ -18,72 +23,147 @@ export type ThemerollerChildrenProps = {
...
+        setThemeResultStore(oneUITheme.result);
+    };
+
+    const handleGenerate = (cssVars) => {
+        oneUITheme.setTheme(themeResultStore.name, cssVars);
+        setThemeResultStore(oneUITheme.result);
+    };
+
+    const handleReset = () => {
+        oneUITheme.setTheme('', {});
```
[the remark this answers is not in the document]
Comment (mukiienko) [reply src/components/Themeroller/Themeroller.tsx:90]: It makes sense!

**Review by ahalyha — APPROVED** [review]

**Pull request merged**